fuse_examples/segmentation/data_source_segmentation.py

```
@@ -0,0 +1,111 @@
+        :param val_split:          validation proportion in case of splitting
+        :param override_partition: specifies if the given partition file is filled with new train/val splits
+        """
+
+
+        # Extract entities
+        # ----------------
+        if partition_file is not None:
+            if train:
+                if override_partition:
+                    train_fn = glob(image_source + '/*')
+                    train_fn.sort()
+
+                    masks_fn = glob(mask_source + '/*')
+                    masks_fn.sort()
+
+                    fn = list(zip(train_fn, masks_fn))
+                    
+                    if len(fn) == 0:
+                        raise Exception('Error detecting input source in FuseDataSourceDefault')
+
+                    if data_shuffle:
+                        # random shuffle the file-list
+                        random.shuffle(fn)
+
+                    # split to train-validation -
+                    n_train = int(len(fn) * (1-val_split))
+
+                    train_fn = fn[:n_train]
+                    val_fn = fn[n_train:]
+                    splits = {'train': train_fn, 'val': val_fn}
+
+                    with open(partition_file, "wb") as pickle_out:
+                        pickle.dump(splits, pickle_out)
+                    sample_descs = train_fn
+                else:
+                    # read from a previous train/test split to evaluate on the same partition
+                    with open(partition_file, "rb") as splits:
+                        repartition = pickle.load(splits)
+                    sample_descs = repartition['train']
+            else:
+                with open(partition_file, "rb") as splits:
+                    repartition = pickle.load(splits)
+                sample_descs = repartition['val']
+        else:
+            # TODO - this option is not clear - if the partition file is not give? do we train 
```

You can throw an error in such a case if you don't want to support it.